pkg/types/operation.go

```diff
@@ -53,19 +53,27 @@ const (
 	FAILED OperationState = "failed"
 )
 
+type RelatedType struct {
+	ID            string            `json:"id"`
+	Criteria      interface{}       `json:"criterion"`
```

Comment:
This is an array of criterias, but I don't remember now why it has to be interface{}. I think it was because it cannot unmarshal it otherwise. 
I will change the name to Criterion

Reply:
If it is array of criterions, probably criteria is better naming for the json tag.